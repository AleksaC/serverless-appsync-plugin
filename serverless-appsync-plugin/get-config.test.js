const getConfig = require('./get-config');

test('authenticationType is missing', () => {
  const run = () => getConfig({}, {});
  expect(run).toThrowErrorMatchingSnapshot();
});

test('serviceRole is missing', () => {
  const run = () =>
    getConfig(
      {
        authenticationType: 'AWS_IAM'
      },
      {}
    );
  expect(run).toThrowErrorMatchingSnapshot();
});

test('returns valid config', () => {
  expect(
    getConfig(
      {
        authenticationType: 'AWS_IAM',
        serviceRole: '1234'
      },
      { region: 'us-east-1' }
    )
  ).toMatchSnapshot();
});
