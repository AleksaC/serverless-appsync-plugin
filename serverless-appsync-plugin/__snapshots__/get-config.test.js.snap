// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authenticationType is missing 1`] = `"appSync property \`authenticationType\` is required."`;

exports[`returns valid config 1`] = `
Object {
  "authenticationType": "AWS_IAM",
  "dataSources": undefined,
  "mappingTemplates": Object {},
  "name": "api",
  "region": "us-east-1",
  "schema": "schema.graphql",
}
`;

exports[`serviceRole is missing 1`] = `"appSync property \`serviceRole\` is required."`;
